universal-login-react/test/integration/http/IPGeolocationService.ts

```diff
@@ -0,0 +1,34 @@
+import chai, {expect} from 'chai';
+import chaiAsPromised from 'chai-as-promised';
+import nock from 'nock';
+import {IPGeolocationService} from '../../../src/integration/http/IPGeolocationService';
+
+chai.use(chaiAsPromised);
+
+describe('IPGeolocationService', function () {
+  const baseUrl = 'http://example.com';
+
+  let service: IPGeolocationService;
+
+  beforeEach(function () {
+    service = new IPGeolocationService(baseUrl, 'ABCD');
+  });
+
+  describe('getCountryCode', function () {
+    it('makes a correct request and returns country code', async function () {
+      nock(baseUrl)
+        .get('/check?access_key=ABCD&fields=country_code')
+        .reply(200, {country_code: 'PL'});
+
+      await expect(service.getCountryCode()).to.eventually.eq('PL');
+    });
+
+    it('throws error in case of invalid response', async function () {
+      nock(baseUrl)
+        .get('/check?access_key=ABCD&fields=country_code')
+        .reply(200, {country_code: null});
+
+      await expect(service.getCountryCode()).to.be.rejected;
```

Comment:
`rejectedWith`?

Reply:
I was working on a fix, will include it in the next PR.